compiler/src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -2517,9 +2517,9 @@ object Parsers {
     def generatorRest(pat: Tree, casePat: Boolean): GenFrom =
       atSpan(startOffset(pat), accept(LARROW)) {
         val checkMode =
-          if (casePat) GenCheckMode.FilterAlways
-          else if sourceVersion.isAtLeast(future) then GenCheckMode.Check
-          else GenCheckMode.FilterNow  // filter for now, to keep backwards compat
+          if casePat then GenCheckMode.FilterAlways
+          else if sourceVersion.isAtLeast(`3.2`) && !sourceVersion.isMigrating then GenCheckMode.Check
```

Comment:
this current code would fail for `future-migration`
```suggestion
          else if sourceVersion.ordinal >= `3.2`.ordinal then GenCheckMode.Check
```
^ though we could perhaps add a variant of `isAtLeast` that does not consider the migration version in the same series

Reply:
however, looking at the old code `future-migration` would use the new behaviour, so I don't think we need the "not migrating" check